internal/services/containers/kubernetes_cluster_auth_resource_test.go

```diff
@@ -142,6 +142,7 @@ func TestAccKubernetesCluster_roleBasedAccessControlDisabled(t *testing.T) {
 }
 
 func TestAccKubernetesCluster_roleBasedAccessControlAAD(t *testing.T) {
+	t.Skip("Azure AD Integration (legacy) (https://aka.ms/aks/aad-legacy) is deprecated, the cluster could not be created with the Azure AD integration (legacy) enabled.")
```

Comment:
Should we also add a deprecation message to the affected properties?

Reply:
The deprecated use is specifying `managed = false`, so I think we couldn't add the deprecated message to the field.

refs: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#managed

Reply:
If AKS doesn't support clusters where `managed = false` i.e. where `client_app_id`, `server_app_id`, `server_app_secret` can be specified for that use case, then those properties (including `managed`) should have deprecation messages and be removed in v4.0

Reply:
Got it! I'll fix it in the next commit.